osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -34,8 +34,11 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
                 return new OsuDifficultyAttributes { Mods = mods, Skills = skills };
 
             double aimRating = Math.Sqrt(skills[0].DifficultyValue()) * difficulty_multiplier;
-            double speedRating = Math.Sqrt(skills[1].DifficultyValue()) * difficulty_multiplier;
-            double flashlightRating = Math.Sqrt(skills[2].DifficultyValue()) * difficulty_multiplier;
+            double aimRatingNoSliders = Math.Sqrt(skills[1].DifficultyValue()) * difficulty_multiplier;
+            double speedRating = Math.Sqrt(skills[2].DifficultyValue()) * difficulty_multiplier;
+            double flashlightRating = Math.Sqrt(skills[3].DifficultyValue()) * difficulty_multiplier;
+
+            double sliderFactor = aimRatingNoSliders / aimRating;
```

Comment:
aimRating can be 0(Examples: [aimRatingZero.txt](https://github.com/ppy/osu/files/7523683/aimRatingZero.txt) or the start of https://osu.ppy.sh/beatmapsets/1335452#osu/2766688), so you can get a division by 0 here.

Reply:
Noticed the same for https://osu.ppy.sh/beatmapsets/774426#osu/1627673
pp at the first slider jumps to MinInt32

https://user-images.githubusercontent.com/5307179/141386225-6cc8732b-20a0-4447-85ea-69819c38f88a.mp4

Reply:
Thanks guys, good catch. A bit embarassing that I didn't think of that...